packages/plugin-apex/test/commands/force/apex/test/run.test.ts

```diff
@@ -1208,4 +1208,53 @@ describe('force:apex:test:run', () => {
     .it('should set exit code as 0 for passing run', () => {
       expect(process.exitCode).to.eql(0);
     });
+
+  test
+    .withOrg({ username: TEST_USERNAME }, true)
+    .loadConfig({
+      root: __dirname
+    })
+    .stub(process, 'cwd', () => projectPath)
+    .stub(TestService.prototype, 'runTestAsynchronous', () => testRunSimple)
+    .stdout()
+    .command([
+      'force:apex:test:run',
+      '--tests',
+      'MyApexTests',
+      '--wait',
+      '20'
+    ])
+    .it('should return human-readable results when the wait argument is passed', ctx => {
+      const result = ctx.stdout;
+      expect(result).to.not.be.empty;
+      expect(result).to.contain('Test Summary');
+      expect(result).to.contain('Test Results');
+      expect(result).to.not.contain('to retrieve test results');
+    });
+
+  test
+    .withOrg({ username: TEST_USERNAME }, true)
+    .loadConfig({
+      root: __dirname
+    })
+    .stub(process, 'cwd', () => projectPath)
+    .stub(TestService.prototype, 'runTestAsynchronous', () => testRunSimple)
+    .stdout()
+    .command([
+      'force:apex:test:run',
+      '--tests',
+      'MyApexTests',
+      '--wait',
+      '20',
+      '--resultformat',
+      'json'
+    ])
+    .it('should return JSON results when the wait argument is passed and the resultformat is JSON', ctx => {
+      const result = ctx.stdout;
+      expect(result).to.not.be.empty;
+      expect(result).to.not.contain('to retrieve test results');
+
+      const obj = JSON.parse(result);
+      expect(obj).to.exist;
```

Comment:
could you add in a test for the follow scenario? `sfdx force:apex:test:run --tests MyTest --wait 20 --json`

Reply:
I left the previous test in (`--resultformat json`) and added a new test (explicitly for `--json`)